### PR TITLE
fix: secret sync show and disable

### PR DESCRIPTION
**Pull request by Elsie4ever**
Originally there is only one secret provider class created after secret sync enablement
Therefore there was an assumption that the first secret provider class is always the default one and got returned immediately
**NOTE: also removed sleep 5 in get key vault client**

however after adding cert management cli, if user ran any of them there will be one default and one for OPC UA connector, therefore we should consider to get all spcs instead of only first returned one

checking existing spc changed to checking existing **spcs**
for `secretsync enable`:  no major behavioral change
for `secretsync disable`: when user decide to disable secretsync, remove all spcs related to instance instead of only first returned one, also instead of only deleting spc, the related secret sync resource will also being deleted
for `secretsync show`: changed to `secretsync list` and will list all existing spc
image

manually tested enable\disable\list all worked as expected

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: azext_edge/edge/providers/orchestration/resources/instances.py [hunk 1]
```diff
@@ -318,15 +319,16 @@ def enable_secretsync(
             return result_spc
 
     def show_secretsync(self, name: str, resource_group_name: str) -> Optional[dict]:
```
Comment (Elsie4ever) [inline azext_edge/edge/providers/orchestration/resources/instances.py:321]: now we might have more than one spc, therefore should we add spc name as a parameter for show or we show just change it to `list`?

Comment (Elsie4ever) [reply azext_edge/edge/providers/orchestration/resources/instances.py]: will switch to list now

File: azext_edge/edge/providers/orchestration/resources/instances.py [hunk 2]
```diff
@@ -346,24 +355,69 @@ def disable_secretsync(
...
+        resources = []
         for resource in cl_resources:
-            if resource["type"].lower() == "microsoft.secretsynccontroller/azurekeyvaultsecretproviderclasses":
+            if resource["type"].lower() == resource_type:
                 resource_id_container = parse_resource_id(resource["id"])
-                return self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
-                    resource_group_name=resource_id_container.resource_group_name,
-                    azure_key_vault_secret_provider_class_name=resource_id_container.resource_name,
-                )
+                if resource_type == "microsoft.secretsynccontroller/azurekeyvaultsecretproviderclasses":
```
Comment (digimaun) [inline azext_edge/edge/providers/orchestration/resources/instances.py:399]: Shouldn't we reference the module constants that were added ?

File: azext_edge/edge/util/az_client.py [hunk 3]
```diff
@@ -196,9 +196,6 @@ def get_keyvault_client(subscription_id: str, keyvault_name: str, **kwargs) -> "
         **kwargs,
     )
 
-    # wait to set the access token
-    sleep(5)
```
Comment (digimaun) [inline azext_edge/edge/util/az_client.py:200]: thanks

File: azext_edge/edge/providers/orchestration/resources/instances.py (hunk 2 above)
Comment (Elsie4ever) [reply azext_edge/edge/providers/orchestration/resources/instances.py:399]: will update!

File: azext_edge/edge/util/az_client.py (hunk 3 above)
Comment (Elsie4ever) [reply azext_edge/edge/util/az_client.py:200]: of course!

**Review by digimaun — APPROVED** [review]

File: azext_edge/edge/_help.py [hunk 4]
```diff
@@ -716,15 +716,15 @@ def load_iotops_help():
...
 
         examples:
-        - name: Show the secret sync config associated with an instance.
+        - name: List the secret sync configs associated with an instance.
           text: >
-            az iot ops secretsync show --name myinstance -g myresourcegroup
+            az iot ops secretsync list --name myinstance -g myresourcegroup
     """
 
     helps[
```
Comment (digimaun) [inline azext_edge/edge/_help.py:730]: In the Disable summary can we append something like all SPCs and related secret sync resources will be deleted ?

Comment (Elsie4ever) [reply azext_edge/edge/_help.py:730]: yeah sounds good, I will update

**Comment by digimaun** [comment]
@Elsie4ever please increment the package version (`constants` module) to 0.8.0b1

File: azext_edge/edge/_help.py (hunk 4 above)
Comment (Elsie4ever) [reply azext_edge/edge/_help.py:730]: image

Comment (Elsie4ever) [reply azext_edge/edge/_help.py:730]: image

**Pull request merged**